File: vortexdbg-api/src/main/java/com/vortexdbg/hook/hookzz/WrapCallback.java
package com.vortexdbg.hook.hookzz;

import com.vortexdbg.Emulator;
import com.vortexdbg.arm.context.RegisterContext;

public abstract class WrapCallback<T extends RegisterContext> {

    public abstract void preCall(Emulator<?> emulator, T ctx, HookEntryInfo info);

    public void postCall(Emulator<?> emulator, T ctx, HookEntryInfo info) {}

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/hook/hookzz/WrapCallback.kt
package com.vortexdbg.hook.hookzz

import com.vortexdbg.Emulator
import com.vortexdbg.arm.context.RegisterContext

abstract class WrapCallback<T : RegisterContext> {

    abstract fun preCall(emulator: Emulator<*>, ctx: T, info: HookEntryInfo)

    open fun postCall(emulator: Emulator<*>, ctx: T, info: HookEntryInfo) {}

}
